### ISSUE TITLE
multiple entries for a VF nic in sunstoine

### ISSUE THREAD

**Issue by atodorov-storpool** [OpenNebula/one#3101]
**Description**

![image](https://user-images.githubusercontent.com/11386997/54685681-17180e80-4b20-11e9-971a-fd1869563a9d.png)

**To Reproduce**
1. Instantiate VM with two nics - first a virtio, second a PCI pass-through.
2. Add third virtio NIC
3. Check the VM's network tab

**Expected behavior**
The VF entry should be displayed only once.

**Details**
 - Affected Component: [Sunstone]
 - Hypervisor: [KVM]
 - Version: [5.8.0]

**Additional context**
The VM's XML:

```
<VM>
  <ID>23</ID>
  <UID>2</UID>
  <GID>0</GID>
  <UNAME>ant</UNAME>
  <GNAME>oneadmin</GNAME>
  <NAME>issuePocVf</NAME>
  <PERMISSIONS>
    <OWNER_U>1</OWNER_U>
    <OWNER_M>1</OWNER_M>
    <OWNER_A>0</OWNER_A>
    <GROUP_U>0</GROUP_U>
    <GROUP_M>0</GROUP_M>
    <GROUP_A>0</GROUP_A>
    <OTHER_U>0</OTHER_U>
    <OTHER_M>0</OTHER_M>
    <OTHER_A>0</OTHER_A>
  </PERMISSIONS>
  <LAST_POLL>1553086761</LAST_POLL>
  <STATE>3</STATE>
  <LCM_STATE>3</LCM_STATE>
  <PREV_STATE>3</PREV_STATE>
  <PREV_LCM_STATE>3</PREV_LCM_STATE>
  <RESCHED>0</RESCHED>
  <STIME>1553084277</STIME>
  <ETIME>0</ETIME>
  <DEPLOY_ID>one-23</DEPLOY_ID>
  <MONITORING>
    <CPU><![CDATA[0.0]]></CPU>
    <DISKRDBYTES><![CDATA[177463836]]></DISKRDBYTES>
    <DISKRDIOPS><![CDATA[7130]]></DISKRDIOPS>
    <DISKWRBYTES><![CDATA[29748736]]></DISKWRBYTES>
    <DISKWRIOPS><![CDATA[3275]]></DISKWRIOPS>
    <DISK_SIZE>
      <ID><![CDATA[0]]></ID>
      <SIZE><![CDATA[1330]]></SIZE>
    </DISK_SIZE>
    <DISK_SIZE>
      <ID><![CDATA[1]]></ID>
      <SIZE><![CDATA[1]]></SIZE>
    </DISK_SIZE>
    <MEMORY><![CDATA[2159264]]></MEMORY>
    <NETRX><![CDATA[1488066]]></NETRX>
    <NETTX><![CDATA[64746]]></NETTX>
    <STATE><![CDATA[a]]></STATE>
  </MONITORING>
  <TEMPLATE>
    <AUTOMATIC_DS_REQUIREMENTS><![CDATA[("CLUSTERS/ID" @> 0)]]></AUTOMATIC_DS_REQUIREMENTS>
    <AUTOMATIC_NIC_REQUIREMENTS><![CDATA[("CLUSTERS/ID" @> 0)]]></AUTOMATIC_NIC_REQUIREMENTS>
    <AUTOMATIC_REQUIREMENTS><![CDATA[(CLUSTER_ID = 0) & !(PUBLIC_CLOUD = YES)]]></AUTOMATIC_REQUIREMENTS>
    <CONTEXT>
      <DISK_ID><![CDATA[1]]></DISK_ID>
      <ETH0_CONTEXT_FORCE_IPV4><![CDATA[]]></ETH0_CONTEXT_FORCE_IPV4>
      <ETH0_DNS><![CDATA[10.2.1.254]]></ETH0_DNS>
      <ETH0_EXTERNAL><![CDATA[]]></ETH0_EXTERNAL>
      <ETH0_GATEWAY><![CDATA[10.2.1.254]]></ETH0_GATEWAY>
      <ETH0_GATEWAY6><![CDATA[]]></ETH0_GATEWAY6>
      <ETH0_IP><![CDATA[10.2.1.142]]></ETH0_IP>
      <ETH0_IP6><![CDATA[]]></ETH0_IP6>
      <ETH0_IP6_PREFIX_LENGTH><![CDATA[]]></ETH0_IP6_PREFIX_LENGTH>
      <ETH0_IP6_ULA><![CDATA[]]></ETH0_IP6_ULA>
      <ETH0_MAC><![CDATA[02:00:0a:02:01:8e]]></ETH0_MAC>
      <ETH0_MASK><![CDATA[255.255.255.0]]></ETH0_MASK>
      <ETH0_MTU><![CDATA[1500]]></ETH0_MTU>
      <ETH0_NETWORK><![CDATA[10.2.1.0]]></ETH0_NETWORK>
      <ETH0_SEARCH_DOMAIN><![CDATA[]]></ETH0_SEARCH_DOMAIN>
      <ETH0_VLAN_ID><![CDATA[]]></ETH0_VLAN_ID>
      <ETH0_VROUTER_IP><![CDATA[]]></ETH0_VROUTER_IP>
      <ETH0_VROUTER_IP6><![CDATA[]]></ETH0_VROUTER_IP6>
      <ETH0_VROUTER_MANAGEMENT><![CDATA[]]></ETH0_VROUTER_MANAGEMENT>
      <ETH2_CONTEXT_FORCE_IPV4><![CDATA[]]></ETH2_CONTEXT_FORCE_IPV4>
      <ETH2_DNS><![CDATA[10.2.81.254]]></ETH2_DNS>
      <ETH2_EXTERNAL><![CDATA[]]></ETH2_EXTERNAL>
      <ETH2_GATEWAY><![CDATA[]]></ETH2_GATEWAY>
      <ETH2_GATEWAY6><![CDATA[]]></ETH2_GATEWAY6>
      <ETH2_IP><![CDATA[]]></ETH2_IP>
      <ETH2_IP6><![CDATA[]]></ETH2_IP6>
      <ETH2_IP6_PREFIX_LENGTH><![CDATA[]]></ETH2_IP6_PREFIX_LENGTH>
      <ETH2_IP6_ULA><![CDATA[]]></ETH2_IP6_ULA>
      <ETH2_MAC><![CDATA[02:00:47:6f:f1:71]]></ETH2_MAC>
      <ETH2_MASK><![CDATA[24]]></ETH2_MASK>
      <ETH2_MTU><![CDATA[1500]]></ETH2_MTU>
      <ETH2_NETWORK><![CDATA[10.2.81.0]]></ETH2_NETWORK>
      <ETH2_SEARCH_DOMAIN><![CDATA[]]></ETH2_SEARCH_DOMAIN>
      <ETH2_VLAN_ID><![CDATA[281]]></ETH2_VLAN_ID>
      <ETH2_VROUTER_IP><![CDATA[]]></ETH2_VROUTER_IP>
      <ETH2_VROUTER_IP6><![CDATA[]]></ETH2_VROUTER_IP6>
      <ETH2_VROUTER_MANAGEMENT><![CDATA[]]></ETH2_VROUTER_MANAGEMENT>
      <NETWORK><![CDATA[YES]]></NETWORK>
      <PCI0_ADDRESS><![CDATA[01:01.0]]></PCI0_ADDRESS>
      <SET_HOSTNAME><![CDATA[issuePocVf]]></SET_HOSTNAME>
      <TARGET><![CDATA[hda]]></TARGET>
    </CONTEXT>
    <CPU><![CDATA[0.1]]></CPU>
    <CPU_MODEL>
      <MODEL><![CDATA[host-passthrough]]></MODEL>
    </CPU_MODEL>
    <DISK>
      <ALLOW_ORPHANS><![CDATA[MIXED]]></ALLOW_ORPHANS>
      <CACHE><![CDATA[none]]></CACHE>
      <CLONE><![CDATA[YES]]></CLONE>
      <CLONE_TARGET><![CDATA[SELF]]></CLONE_TARGET>
      <CLUSTER_ID><![CDATA[0]]></CLUSTER_ID>
      <DATASTORE><![CDATA[default]]></DATASTORE>
      <DATASTORE_ID><![CDATA[1]]></DATASTORE_ID>
      <DEV_PREFIX><![CDATA[sd]]></DEV_PREFIX>
      <DISCARD><![CDATA[unmap]]></DISCARD>
      <DISK_ID><![CDATA[0]]></DISK_ID>
      <DISK_SNAPSHOT_TOTAL_SIZE><![CDATA[0]]></DISK_SNAPSHOT_TOTAL_SIZE>
      <DISK_TYPE><![CDATA[BLOCK]]></DISK_TYPE>
      <DRIVER><![CDATA[raw]]></DRIVER>
      <IMAGE><![CDATA[CentOS-7-1810-contextualized]]></IMAGE>
      <IMAGE_ID><![CDATA[3]]></IMAGE_ID>
      <IMAGE_STATE><![CDATA[2]]></IMAGE_STATE>
      <IMAGE_UNAME><![CDATA[ant]]></IMAGE_UNAME>
      <IO><![CDATA[native]]></IO>
      <LN_TARGET><![CDATA[NONE]]></LN_TARGET>
      <ORDER><![CDATA[1]]></ORDER>
      <ORIGINAL_SIZE><![CDATA[10240]]></ORIGINAL_SIZE>
      <READONLY><![CDATA[NO]]></READONLY>
      <SAVE><![CDATA[NO]]></SAVE>
      <SIZE><![CDATA[10240]]></SIZE>
      <SOURCE><![CDATA[mrs-ds-1/mrs-img-3]]></SOURCE>
      <TARGET><![CDATA[sda]]></TARGET>
      <TM_MAD><![CDATA[storpool]]></TM_MAD>
      <TYPE><![CDATA[BLOCK]]></TYPE>
    </DISK>
    <FEATURES>
      <ACPI><![CDATA[yes]]></ACPI>
      <APIC><![CDATA[yes]]></APIC>
      <GUEST_AGENT><![CDATA[yes]]></GUEST_AGENT>
      <PAE><![CDATA[yes]]></PAE>
      <VIRTIO_SCSI_QUEUES><![CDATA[1]]></VIRTIO_SCSI_QUEUES>
    </FEATURES>
    <GRAPHICS>
      <LISTEN><![CDATA[0.0.0.0]]></LISTEN>
      <PORT><![CDATA[5923]]></PORT>
      <TYPE><![CDATA[VNC]]></TYPE>
    </GRAPHICS>
    <INPUT>
      <BUS><![CDATA[usb]]></BUS>
      <TYPE><![CDATA[tablet]]></TYPE>
    </INPUT>
    <MEMORY><![CDATA[2048]]></MEMORY>
    <NIC>
      <AR_ID><![CDATA[0]]></AR_ID>
      <BRIDGE><![CDATA[br61]]></BRIDGE>
      <BRIDGE_TYPE><![CDATA[linux]]></BRIDGE_TYPE>
      <CLUSTER_ID><![CDATA[0]]></CLUSTER_ID>
      <IP><![CDATA[10.2.1.142]]></IP>
      <MAC><![CDATA[02:00:0a:02:01:8e]]></MAC>
      <MODEL><![CDATA[virtio]]></MODEL>
      <NAME><![CDATA[NIC0]]></NAME>
      <NETWORK><![CDATA[net-10.2.1]]></NETWORK>
      <NETWORK_ID><![CDATA[1]]></NETWORK_ID>
      <NIC_ID><![CDATA[0]]></NIC_ID>
      <SECURITY_GROUPS><![CDATA[0]]></SECURITY_GROUPS>
      <TARGET><![CDATA[one-23-0]]></TARGET>
      <VN_MAD><![CDATA[bridge]]></VN_MAD>
    </NIC>
    <NIC>
      <AR_ID><![CDATA[0]]></AR_ID>
      <BRIDGE><![CDATA[br281]]></BRIDGE>
      <BRIDGE_TYPE><![CDATA[linux]]></BRIDGE_TYPE>
      <CLUSTER_ID><![CDATA[0]]></CLUSTER_ID>
      <MAC><![CDATA[02:00:47:6f:f1:71]]></MAC>
      <MODEL><![CDATA[virtio]]></MODEL>
      <MTU><![CDATA[9000]]></MTU>
      <NAME><![CDATA[NIC2]]></NAME>
      <NETWORK><![CDATA[net-10.2.81]]></NETWORK>
      <NETWORK_ID><![CDATA[4]]></NETWORK_ID>
      <NETWORK_UNAME><![CDATA[ant]]></NETWORK_UNAME>
      <NIC_ID><![CDATA[2]]></NIC_ID>
      <PHYDEV><![CDATA[spbond0]]></PHYDEV>
      <SECURITY_GROUPS><![CDATA[0]]></SECURITY_GROUPS>
      <TARGET><![CDATA[one-23-2]]></TARGET>
      <VLAN_ID><![CDATA[281]]></VLAN_ID>
      <VN_MAD><![CDATA[802.1Q]]></VN_MAD>
    </NIC>
    <NIC_DEFAULT>
      <MODEL><![CDATA[virtio]]></MODEL>
    </NIC_DEFAULT>
    <OS>
      <ARCH><![CDATA[x86_64]]></ARCH>
      <BOOT><![CDATA[disk0]]></BOOT>
      <MACHINE><![CDATA[pc]]></MACHINE>
    </OS>
    <PCI>
      <ADDRESS><![CDATA[0000:d8:00:5]]></ADDRESS>
      <BUS><![CDATA[d8]]></BUS>
      <CLASS><![CDATA[0200]]></CLASS>
      <DEVICE><![CDATA[1018]]></DEVICE>
      <DOMAIN><![CDATA[0000]]></DOMAIN>
      <FUNCTION><![CDATA[5]]></FUNCTION>
      <MODEL><![CDATA[virtio]]></MODEL>
      <NAME><![CDATA[NIC1]]></NAME>
      <NIC_ID><![CDATA[1]]></NIC_ID>
      <PCI_ID><![CDATA[0]]></PCI_ID>
      <SLOT><![CDATA[00]]></SLOT>
      <TYPE><![CDATA[NIC]]></TYPE>
      <VENDOR><![CDATA[15b3]]></VENDOR>
      <VM_ADDRESS><![CDATA[01:01.0]]></VM_ADDRESS>
      <VM_BUS><![CDATA[0x01]]></VM_BUS>
      <VM_DOMAIN><![CDATA[0x0000]]></VM_DOMAIN>
      <VM_FUNCTION><![CDATA[0]]></VM_FUNCTION>
      <VM_SLOT><![CDATA[0x01]]></VM_SLOT>
    </PCI>
    <SECURITY_GROUP_RULE>
      <PROTOCOL><![CDATA[ALL]]></PROTOCOL>
      <RULE_TYPE><![CDATA[OUTBOUND]]></RULE_TYPE>
      <SECURITY_GROUP_ID><![CDATA[0]]></SECURITY_GROUP_ID>
      <SECURITY_GROUP_NAME><![CDATA[default]]></SECURITY_GROUP_NAME>
    </SECURITY_GROUP_RULE>
    <SECURITY_GROUP_RULE>
      <PROTOCOL><![CDATA[ALL]]></PROTOCOL>
      <RULE_TYPE><![CDATA[INBOUND]]></RULE_TYPE>
      <SECURITY_GROUP_ID><![CDATA[0]]></SECURITY_GROUP_ID>
      <SECURITY_GROUP_NAME><![CDATA[default]]></SECURITY_GROUP_NAME>
    </SECURITY_GROUP_RULE>
    <TEMPLATE_ID><![CDATA[3]]></TEMPLATE_ID>
    <TM_MAD_SYSTEM><![CDATA[storpool]]></TM_MAD_SYSTEM>
    <VCPU><![CDATA[1]]></VCPU>
    <VMID><![CDATA[23]]></VMID>
  </TEMPLATE>
  <USER_TEMPLATE>
    <HYPERVISOR><![CDATA[kvm]]></HYPERVISOR>
    <INPUTS_ORDER><![CDATA[]]></INPUTS_ORDER>
    <LOGO><![CDATA[images/logos/centos.png]]></LOGO>
    <MEMORY_UNIT_COST><![CDATA[MB]]></MEMORY_UNIT_COST>
  </USER_TEMPLATE>
  <HISTORY_RECORDS>
    <HISTORY>
      <OID>23</OID>
      <SEQ>0</SEQ>
      <HOSTNAME>a2</HOSTNAME>
      <HID>1</HID>
      <CID>0</CID>
      <STIME>1553084296</STIME>
      <ETIME>1553084580</ETIME>
      <VM_MAD><![CDATA[kvm]]></VM_MAD>
      <TM_MAD><![CDATA[storpool]]></TM_MAD>
      <DS_ID>0</DS_ID>
      <PSTIME>1553084296</PSTIME>
      <PETIME>1553084306</PETIME>
      <RSTIME>1553084306</RSTIME>
      <RETIME>1553084580</RETIME>
      <ESTIME>0</ESTIME>
      <EETIME>0</EETIME>
      <ACTION>23</ACTION>
      <UID>2</UID>
      <GID>0</GID>
      <REQUEST_ID>624</REQUEST_ID>
    </HISTORY>
    <HISTORY>
      <OID>23</OID>
      <SEQ>1</SEQ>
      <HOSTNAME>a2</HOSTNAME>
      <HID>1</HID>
      <CID>0</CID>
      <STIME>1553084580</STIME>
      <ETIME>0</ETIME>
      <VM_MAD><![CDATA[kvm]]></VM_MAD>
      <TM_MAD><![CDATA[storpool]]></TM_MAD>
      <DS_ID>0</DS_ID>
      <PSTIME>0</PSTIME>
      <PETIME>0</PETIME>
      <RSTIME>1553084580</RSTIME>
      <RETIME>0</RETIME>
      <ESTIME>0</ESTIME>
      <EETIME>0</EETIME>
      <ACTION>0</ACTION>
      <UID>-1</UID>
      <GID>-1</GID>
      <REQUEST_ID>-1</REQUEST_ID>
    </HISTORY>
  </HISTORY_RECORDS>
</VM>
```

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by tinova** [comment]
needs testing

**Comment by atodorov-storpool** [comment]
Just to report that I've backported the patch to 5.8.1 and it works!

**Comment by atodorov-storpool** [comment]
Well just discovered glitch in the Update Config -> OS -> Boot -> Boot Order:

![Screenshot from 2019-05-13 14-10-37](https://user-images.githubusercontent.com/11386997/57617347-4596f080-7589-11e9-8d09-c2c986b4c640.png)

The VM in question has two VF and two virtio nics:
![Screenshot from 2019-05-13 14-10-51](https://user-images.githubusercontent.com/11386997/57617392-60696500-7589-11e9-87a1-3e282081df4e.png)

Do you prefer to open another issue for it?
